operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSource.java

```diff
@@ -1,10 +1,11 @@
 package io.javaoperatorsdk.operator.processing.event.source;
 
+import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.javaoperatorsdk.operator.processing.LifecycleAware;
-import io.javaoperatorsdk.operator.processing.event.EventHandler;
 
-public interface EventSource extends LifecycleAware {
+public interface EventSource<P extends HasMetadata> extends LifecycleAware {
 
-  void setEventHandler(EventHandler eventHandler);
+  void setEventRegistry(EventSourceRegistry<P> registry);
```

Comment:
Here the method naming is not good. But don't get this change, why should an event source be aware of the whole registry?

Reply:
Good catch on the name. Why wouldn't an event source be aware of where it's registered? The idea here is that knowing the registry allows an `EventSource` to query the registry for other resources without having to hold a reference to the other event sources.